v2/internal/reconcilers/arm/arm_client_cache.go

```diff
@@ -82,37 +85,96 @@ func (c *armClientCache) lookup(key string) (*armClient, bool) {
 
 // GetClient finds and returns a client and credential to be used for a given resource
 func (c *armClientCache) GetClient(ctx context.Context, obj genruntime.ARMMetaObject) (*genericarmclient.GenericClient, string, error) {
+
+	client, err := c.checkPerResourceSecret(ctx, obj)
```

Comment:
minor, naming: Would these `checkX` methods be more clearly named `getClientFromX` or maybe just `getXClient`?

So `getPerResourceClient`, `getNamespacedClient`?

Obviously very minor, ignore if you don't think that's clearer.